runtime/meta.go

```diff
@@ -0,0 +1,138 @@
+			Time: config.Cfg.Runtime.DeployedAt,
+		},
+	}
+}
+
+func init() {
+	if config.Cfg != nil {
+		// We don't need to check for errors, as we already check it's parsable during the config.ParseConfig initialization
+		apiBaseURL, _ = url.Parse(config.Cfg.Runtime.APIBaseURL)
+	}
+}
+
+// AppMetadata contains metadata about the running Encore application.
+type AppMetadata struct {
+	// The application ID, if the application is not linked to the Encore platform this will be an empty string.
+	//
+	// To link to the Encore platform run `encore app link` from your terminal in the root directory of the Encore app.
+	AppID string
+
+	// The base URL which can be used to call the API of this running application.
+	//
+	// For local development it is "http://localhost:<port>", typically "http://localhost:4000".
+	//
+	// If a custom domain is used for this environment it is returned here, but note that
+	// changes only take effect at the time of deployment while custom domains can be updated at any time.
+	APIBaseURL url.URL
+
+	// Information about the environment the app is running in.
+	Environment EnvironmentMeta
+
+	// Information about the running binary itself.
+	Build BuildMeta
+
+	// Information about this deployment of the binary
+	Deploy DeployMeta
+}
+
+type EnvironmentMeta struct {
+	// The name of environment that this application.
+	// For local development it is "local".
+	Name string
+
+	// The type of environment is this application running in
+	// For local development this will be EnvLocal
+	Type EnvironmentType
+
+	// The cloud that this environment is running on
+	// For local development this is CloudLocal
+	Cloud CloudProvider
+}
+
+type BuildMeta struct {
+	// The git commit that formed the base of this build.
+	Revision string
+
+	// true if there were uncommitted changes on top of the Commit.
+	UncommittedChanges bool
+}
+
+type DeployMeta struct {
+	// The deployment ID created by the Encore Platform.
+	ID string
+
+	// The time the Encore Platform deployed this build to the environment.
+	Time time.Time
+}
+
+// EnvironmentType represents the type of environment.
+//
+// For more information on environment types see https://encore.dev/docs/deploy/environments#environment-types
+//
+// Additional environment types may be added in the future.
+type EnvironmentType string
+
+const (
+	// EnvProduction represents a production environment.
+	EnvProduction EnvironmentType = "production"
+
+	// EnvDevelopment represents a long-lived cloud-hosted, non-production environment, such as test environments.
+	EnvDevelopment EnvironmentType = "development"
+
+	// EnvEphemeral represents short-lived cloud-hosted, non-production environments, such as preview environments
+	// that only exist while a particular pull request is open.
+	EnvEphemeral EnvironmentType = "ephemeral"
+
+	// EnvLocal represents the local development environment when using 'encore run' or `encore test`.
+	EnvLocal EnvironmentType = "local"
```

Comment:
Thoughts on removing the `Env` prefix on these? They seem specific enough to be unlikely to clash with something else down the line

Reply:
There are two `Local`'s already in this package, `CloudLocal` and `EnvLocal`.

I'd prefer to keep the prefix tbh as then it doesn't become confusing what `Local` and `CloudLocal` are